setup.py

```diff
@@ -116,8 +116,9 @@ def run(self):
             'sphinx_rtd_theme = sphinx_rtd_theme',
         ]
     },
+    python_requires='>=2.7,!=3.0.*,!=3.1.*,!=3.2.*,!=3.3.*',
```

Comment:
If we need `python_requires` of `>=2.7` here to begin the process of deprecating 2.7, then I think we should isolate that from deprecating other python versions. Currently, it doesn't seem there is a reason for us to drop support for any of the lesser supported python3 minor versions.

Reply:
These python3 versions have been dropped since  sphinx 1.5: https://www.sphinx-doc.org/en/master/changes.html#release-1-5-released-dec-5-2016